source/apiVolontaria/apiVolontaria/views.py

```diff
@@ -97,7 +98,19 @@ class Users(generics.ListCreateAPIView):
     serializer_class = serializers.UserBasicSerializer
 
     def get_queryset(self):
-        return User.objects.all()
+
+        queryset = User.objects.all()
+        search = self.request.query_params.get('search', None)
+
+        if search:
+            queryset = queryset.filter(
+                Q(username__istartswith=search) |
+                Q(first_name__istartswith=search) |
+                Q(last_name__istartswith=search) |
+                Q(email__istartswith=search)
+            )
+
+        return queryset
```

Comment:
DRF already give us a custom search: https://www.django-rest-framework.org/api-guide/filtering/#searchfilter

It's more easy and less verbose :)

Reply:
Nice, always learning new tricks, thanks for that !